source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -150,7 +150,10 @@ int DefaultCertValidator::initializeSslContexts(std::vector<SSL_CTX*> contexts,
     if (!cert_validation_config->subjectAltNameMatchers().empty()) {
       for (const envoy::extensions::transport_sockets::tls::v3::SubjectAltNameMatcher& matcher :
            cert_validation_config->subjectAltNameMatchers()) {
-        subject_alt_name_matchers_.emplace_back(createStringSanMatcher(matcher));
+        auto san_matcher = createStringSanMatcher(matcher);
+        if (san_matcher != nullptr) {
```

Comment:
Should this throw instead of silently ignoring the error?

Reply:
Yeah I think so, seems like validator is allowed to throw. I changed it to throw an exception instead of skipping.